src/realm/object-store/dictionary.hpp

```diff
@@ -42,8 +50,24 @@ struct DictionaryChangeSet {
     // Keys of objects/values which were modified
     std::vector<Mixed> modifications;
```

Comment:
Now that we have the `add_*` methods to build up the change set, should we mark the vectors as const to prevent them from being filled otherwise?

Reply:
That would require that we flood the code with const_cast when we want to update internally. Could be done - but I think the cost in readability is not justified by the increased safety.

Reply:
... and you could say that the user is free do do what he wants with the vectors.

Reply:
For sure - users gets a copy and they're free to mangle them as they will. I was just under the assumption that we only mutate the vectors through the `add_*` methods and if we mark the vectors as const the only place we need to cost_cast them is inside the `add` method. And the intention isn't to prevent the users from mutating them, but to help ourselves to remember to not mutate the vectors directly when we build up the changeset. But if the cost of this is too many ugly casts, then I agree it's not worth it.